### PR TITLE
Feature/socket connection

### PR DESCRIPTION
**Pull request by IILTCC**
finished connection from telemetry device to simulator

### CODE REVIEW

File: telemetry_device/TelemetryDevice.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        const int PORT = 50000;
+        TcpClient simulator;
+        NetworkStream stream;
+        private Dictionary<IcdTypes, (Type, string)> IcdDictionary;
+        const string FILE_TYPE = ".json";
+        const string REPO_PATH = "../../../icd_repo/";
+        public TelemetryDevice()
+        {
+            this.IcdDictionary = new Dictionary<IcdTypes, (Type, string)>();
+            IcdDictionary.Add(IcdTypes.FlightBoxDown, (typeof(FlightBoxIcd), "FlightBoxUpIcd"));
```
Comment (eyal-otiker) [inline telemetry_device/TelemetryDevice.cs:25]: same comment in simulator for "FlightBoxUpIcd"

File: telemetry_device/TelemetryDevice.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            IcdDictionary.Add(IcdTypes.FlightBoxUp, (typeof(FlightBoxIcd), "FlightboxUpIcd"));
```
Comment (eyal-otiker) [inline telemetry_device/TelemetryDevice.cs:26]: same

File: telemetry_device/TelemetryDevice.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            IcdDictionary.Add(IcdTypes.FiberBoxDown, (typeof(FiberBoxIcd), "FiberBoxDownIcd"));
```
Comment (eyal-otiker) [inline telemetry_device/TelemetryDevice.cs:27]: same

File: telemetry_device/TelemetryDevice.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            socket.Start(1);
+            simulator = await socket.AcceptTcpClientAsync();
+            stream = simulator.GetStream();
+        }
+        public async Task RunAsync()
+        {
+            await ConnectAsync();
+
+            Task.Run(() => { ListenForPackets(); });
+            await Task.Delay(-1);
```
Comment (eyal-otiker) [inline telemetry_device/TelemetryDevice.cs:43]: why -1?

File: telemetry_device/TelemetryDevice.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        public void ProccessPackets(IcdTypes type, byte[]packet)
+        {
+            Type genericIcdType = typeof(IcdPacketDecryptor<>).MakeGenericType(IcdDictionary[type].Item1);
```
Comment (eyal-otiker) [inline telemetry_device/TelemetryDevice.cs:47]: why item1?

File: telemetry_device/TelemetryDevice.cs (hunk 4 above)
Comment (IILTCC) [reply telemetry_device/TelemetryDevice.cs:43]: without it the whole program closes after connecting to the simulator, it is the same as Task.WaitAll

Comment (eyal-otiker) [reply telemetry_device/TelemetryDevice.cs:44]: Ok, add comment here

File: telemetry_device/TelemetryDevice.cs (hunk 5 above)
Comment (eyal-otiker) [reply telemetry_device/TelemetryDevice.cs:48]: You don't need touple for dictionary, change that key of dictionary will be name of IcdType and key will be the type like this: <IcdType, Type>

**Pull request merged**